tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -0,0 +1,756 @@
+	}
+	ipAddressProperties := *(result.PublicIPAddressPropertiesFormat)
+	ipAddress = *(ipAddressProperties.IPAddress)
+	return
+}
+
+// constructWinRMConfig constructs the winrm Listeners for the instance for now it will configure only one listener, but this may
+// scale to multiple listeners depending on the use case. By default they listen via Http protocol.
+func constructWinRMConfig() (winRMConfig *compute.WinRMConfiguration) {
+	winRMListeners := &[]compute.WinRMListener{
+		{
+			Protocol: "Http",
+		},
+	}
+	winRMConfig = &compute.WinRMConfiguration{
+		Listeners: winRMListeners,
+	}
+	return winRMConfig
+}
+
+// constructOSProfile constructs the OS Profile for the creation of windows instance. The OS Profile consists of information
+// such as configuring remote management listeners, instance access setup.
+func (az *AzureProvider) constructOSProfile(ctx context.Context) (osProfile *compute.OSProfile, vmName, password string) {
+	instanceName := "windows-" + randomString(5)
+	adminUserName := "windows"
+	adminPassword := randomPasswordString(12)
+	computeWinRMConfig := constructWinRMConfig()
+	nodeLocation := *(az.getvnetProfile(ctx).Location)
+	var TimeZone string
+	if nodeLocation == "centralus" {
+		TimeZone = "Central Standard Time"
+	}
+	osProfile = &compute.OSProfile{
+		ComputerName:  to.StringPtr(instanceName),
+		AdminUsername: to.StringPtr(adminUserName),
+		AdminPassword: to.StringPtr(adminPassword),
+		WindowsConfiguration: &compute.WindowsConfiguration{
+			ProvisionVMAgent:       to.BoolPtr(true),
+			EnableAutomaticUpdates: to.BoolPtr(false),
+			TimeZone:               to.StringPtr(TimeZone),
+			WinRM:                  computeWinRMConfig,
+		},
+	}
+	return osProfile, instanceName, adminPassword
+}
+
+// constructNetworkProfile constructs the Network Profile for the instance to be created.
+// The network profile consists of information such as nic ID that to be attached to the instance
+// and if multiple ID's need to be configured select which one need to be primary.
+func (az *AzureProvider) constructNetworkProfile(ctx context.Context,
+	vmName string) (networkProfile *compute.NetworkProfile, err error) {
+	vnetProfile := az.getvnetProfile(ctx)
+	vnetName := *(vnetProfile.Name)
+	subnetList := *(vnetProfile.VirtualNetworkPropertiesFormat.Subnets)
+	var index int
+
+	for i, subnet := range subnetList {
+		response, _ := regexp.MatchString("\\bworker\\b", *(subnet.Name))
+		if response {
+			index = i
+			break
+		}
+	}
+
+	subnetName := *(subnetList[index].Name)
+	subnetProperties := *(subnetList[index].SubnetPropertiesFormat)
+	nsgStruct := *(subnetProperties.NetworkSecurityGroup)
+	vmRandString := strings.Split(vmName, "-")[1]
+	var nsgName string
+	if verifyForResource(*(nsgStruct.ID)) {
+		nsgID := *(nsgStruct.ID)
+		nsgName = extractResourceName(nsgID)
+		err := az.createSecurityGroup(ctx, nsgName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create security group rules: %v", err)
+		}
+	} else {
```

Comment:
Add some comments here like `if security group doesn't exist, create a new one` or something like that.

Reply:
removed the use of regex here, implemented the same with strings.Contains()